filament/backend/src/metal/MetalDriver.mm

```diff
@@ -2275,29 +2275,24 @@
 void MetalDriver::mapBufferR(MemoryMappedBufferHandle mmbh,
         BufferObjectHandle boh, size_t offset,
         size_t size, MapBufferAccessFlags access, utils::ImmutableCString&& tag) {
-    construct_handle<MetalMemoryMappedBuffer>(mmbh, boh, offset, size, access);
+    construct_handle<MetalMemoryMappedBuffer>(mmbh, mHandleAllocator, boh, offset, size, access);
```

Comment:
You can do the `handle_cast` here and pass in the `MetalBufferObject*` directly, that way we don't need to pass in the whole handle allocator. You can also check that `boh` is a valid handle here first.